proto/config/appconfig.proto

```diff
@@ -203,6 +203,13 @@ message AppInstanceConfig {
 
   // SnapshotConfig is used to track the snapshots of the app instance
   SnapshotConfig snapshot = 22;
+
+  message PatchEnvelopeRef {
+    // one of them should be non-empty
+    // non-empty name overrides the id
+    string name = 23;
+    string id = 24;
+  }
```

Comment:
Typically we define messages at the base level and use it inside another message. Here you appear to define it inside AppInstanceConfig and not using it at all! Am I reading this correctly?

Reply:
My bad, we will use `PatchEnvelopeRef` in `AppInstance` and `AppInstStatusMsg`, will add it right now

Reply:
My bad, we are instantiating `PatchEnvelopeRef` inside `AppInstanceConfig`